modules/openapi-generator/src/main/resources/typescript/api/api.mustache

```diff
@@ -0,0 +1,212 @@
+// TODO: better import syntax?
+import { BaseAPIRequestFactory, RequiredError } from './baseapi';
+import {Configuration} from '../configuration';
+import { RequestContext, HttpMethod, ResponseContext, HttpFile} from '../http/http';
+{{#platforms}}
+{{#node}}
+import * as FormData from "form-data";
+{{/node}}
+{{/platforms}}
+import {ObjectSerializer} from '../models/ObjectSerializer';
+import {ApiException} from './exception';
+import {isCodeInRange} from '../util';
+
+{{#imports}}
+import { {{classname}} } from '..{{filename}}';
+{{/imports}}
+{{#operations}}
+
+/**
+ * {{#description}}{{{description}}}{{/description}}{{^description}}no description{{/description}}
+ */
+export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
+	
+	{{#operation}}
+    /**
+     {{#notes}}
+     * {{&notes}}
+     {{/notes}}
+     {{#summary}}
+     * {{&summary}}
+     {{/summary}}
+     {{#allParams}}
+     * @param {{paramName}} {{description}}
+     {{/allParams}}
+     */
+    public {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}options?: Configuration): RequestContext {
```

Comment:
I get your point, but personally I would prefer the current structure.
Ordering or naming problems where never a problem in my experience.
Also, calling with an object would make usage slightly more complicated, especially with few arguments.

Reply:
combining the params into one single object can be very helpful when e.g. the api is refactored or extended, so that the compiler will immediately spot naming mismatches, even if the types would match.
I guess it would be best to provide an option for that, see e.g. 
https://github.com/OpenAPITools/openapi-generator/pull/6288
https://github.com/OpenAPITools/openapi-generator/pull/4479

Reply:
That sounds good.

Reply:
@bodograumann Are you already working on this? If not I'll give it a shot tomorrow.

Reply:
No, I'm not working on this. Please go ahead :-)